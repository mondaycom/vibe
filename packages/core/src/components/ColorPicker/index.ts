export { default as ColorPicker, type ColorPickerProps } from "./ColorPicker";
export {
  default as ColorPickerContent,
  type ColorPickerContentProps
} from "./components/ColorPickerContent/ColorPickerContent";

export * from "./ColorPicker.types";
