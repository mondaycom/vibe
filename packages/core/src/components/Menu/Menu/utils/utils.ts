import { type ReactElement } from "react";
import { type MenuChild } from "../MenuConstants";

export function isMenuChildSelectable(child: MenuChild): boolean {
  return !!child.type.isSelectable && !child.props.disabled;
}

export const generateMenuItemId = (menuId: string, child: ReactElement, index: number) => {
  return (child.props as { id: string }).id || `${menuId}-item-${index}`;
};
