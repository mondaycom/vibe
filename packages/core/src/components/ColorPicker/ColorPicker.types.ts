import { type CONTENT_COLORS_VALUES } from "../../utils/colors-vars-map";

export type ColorShapes = "square" | "circle";

export type ColorPickerSizes = "small" | "medium" | "large";

export type ColorPickerValueOnly = CONTENT_COLORS_VALUES | string;
export type ColorPickerArrayValueOnly = CONTENT_COLORS_VALUES[] | string[];
export type ColorPickerValue = ColorPickerValueOnly | ColorPickerArrayValueOnly;
