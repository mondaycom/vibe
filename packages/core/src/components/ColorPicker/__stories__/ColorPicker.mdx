import { Canvas, Meta } from "@storybook/blocks";
import * as ColorPickerStories from "./ColorPicker.stories";

<Meta of={ColorPickerStories} />

# ColorPicker

ColorPicker is our component for selecting our content colors

<Canvas of={ColorPickerStories.Overview} />

### Import

```js
import { ColorPicker } from "@vibe/core";
```

## Props

<PropsTable />

## Variants

### With Indicator

<Canvas of={ColorPickerStories.WithIndicator} />

### Text Indication

<Canvas of={ColorPickerStories.TextIndication} />

### Selected

<Canvas of={ColorPickerStories.Selected} />

### No color

Ability to revert to initial color

<Canvas of={ColorPickerStories.NoColor} />

### Selected icon

Selected colors indication, when using multi-selection

<Canvas of={ColorPickerStories.SelectedIcon} />

### Shapes

The `ColorPicker` supports multiple shapes

<Canvas of={ColorPickerStories.Shapes} />
