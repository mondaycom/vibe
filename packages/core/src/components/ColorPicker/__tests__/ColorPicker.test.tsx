import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, act, screen } from "@testing-library/react";
import { difference as _difference } from "lodash-es";
import ColorPicker from "../ColorPicker";
import { contentColors } from "../../../utils/colors-vars-map";
import ColorPickerColorsGrid from "../components/ColorPickerContent/ColorPickerColorsGrid";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<ColorPicker />).toJSON();
  expect(tree).toMatchSnapshot();
});

jest.useFakeTimers();

describe("ColorPicker", () => {
  it("Should call onSave with color clicked value", () => {
    const colorToClick = contentColors[0];
    let clickedColorValue;
    const onSaveMock = jest.fn();

    const { getByLabelText } = render(
      <ColorPicker
        onSave={color => {
          onSaveMock();
          clickedColorValue = color;
        }}
      />
    );

    const colorElementToClick = getByLabelText(colorToClick);

    act(() => {
      fireEvent.click(colorElementToClick);
    });

    expect(onSaveMock.mock.calls.length).toBe(1);
    expect(clickedColorValue).toStrictEqual([colorToClick]);
  });

  it("Should call onSave with multiselect colors clicked values", () => {
    const colorToClick = contentColors[0];
    let clickedColorValue;
    const onSaveMock = jest.fn();

    const { getByLabelText } = render(
      <ColorPicker
        isMultiselect
        onSave={colors => {
          onSaveMock();
          clickedColorValue = colors;
        }}
      />
    );

    const colorElementToClick = getByLabelText(colorToClick);

    act(() => {
      fireEvent.click(colorElementToClick);
    });

    expect(onSaveMock.mock.calls.length).toBe(1);
    expect(clickedColorValue).toStrictEqual([colorToClick]);
  });

  it("should render only colors not in black list", () => {
    const blackListColors = contentColors.slice(0, 3);
    const restOfColors = _difference(contentColors, blackListColors);

    const { getByLabelText, queryByLabelText } = render(<ColorPicker isBlackListMode colorsList={blackListColors} />);

    blackListColors.forEach(color => expect(queryByLabelText(color)).toBeNull());
    const restOfColorsElements = restOfColors.map(color => getByLabelText(color));

    expect(restOfColorsElements.length).toBe(restOfColors.length);
  });

  it("should render only colors in white list", () => {
    const whiteListColors = contentColors.slice(0, 3);
    const restOfColors = _difference(contentColors, whiteListColors);
    const { getByLabelText, queryByLabelText } = render(
      <ColorPicker isBlackListMode={false} colorsList={whiteListColors} />
    );

    const whiteListColorsElements = whiteListColors.map(color => getByLabelText(color));
    restOfColors.forEach(color => expect(queryByLabelText(color)).toBeNull());

    expect(whiteListColorsElements.length).toBe(whiteListColors.length);
  });

  it("should render all colors if forceUseRawColorList is true and isBlackListMode is false", () => {
    const colorList = ["#abcdef", "#123456", "#234567"];
    const { getByLabelText } = render(<ColorPicker colorsList={colorList} forceUseRawColorList={true} />);

    const colorsElements = colorList.map(color => getByLabelText(color));

    expect(colorsElements.length).toBe(colorsElements.length);
  });

  it("should render all colors if forceUseRawColorList is true, even if isBlackListMode is true", () => {
    const colorList = ["#abcdef", "#123456", "#234567"];
    const { getByLabelText } = render(
      <ColorPicker colorsList={colorList} forceUseRawColorList={true} isBlackListMode={true} />
    );

    const colorsElements = colorList.map(color => getByLabelText(color));

    expect(colorsElements.length).toBe(colorsElements.length);
  });

  it("should render tooltip with color name if showColorNameTooltip is true", () => {
    const colorPicker = render(<ColorPicker showColorNameTooltip />);
    const colorName = "done-green";
    const colorNameTooltip = "Done Green";

    const component = colorPicker.getByLabelText(colorName);
    act(() => {
      fireEvent.mouseOver(component);
    });
    jest.advanceTimersByTime(1000);
    const content = screen.getByText(colorNameTooltip);
    expect(content).toBeTruthy();
  });

  it("should render tooltip with the color name if showColorNameTooltip is true and forceUseRawColorList is true", () => {
    const colorName = "done-green";
    const colorNameTooltip = "Done Green";
    const colorList = [colorName];
    const colorPicker = render(<ColorPicker colorsList={colorList} showColorNameTooltip forceUseRawColorList />);

    const component = colorPicker.getByLabelText(colorName);
    act(() => {
      fireEvent.mouseOver(component);
    });
    jest.advanceTimersByTime(1000);
    const content = screen.queryByText(colorNameTooltip);
    expect(content).toBeTruthy();
  });

  it("should render tooltip with tooltipContentByColor value if tooltipContentByColor of the color exist", () => {
    const colorName = "done-green";
    const colorNameTooltip = "Done Green";
    const contentByColorTooltip = "Custom tooltip";
    const tooltipContentByColor = { "done-green": contentByColorTooltip };
    const colorsToRender = [colorName];

    const colorPicker = render(
      <ColorPickerColorsGrid
        colorsToRender={colorsToRender}
        tooltipContentByColor={tooltipContentByColor}
        showColorNameTooltip
      />
    );

    const component = colorPicker.getByLabelText(colorName);
    act(() => {
      fireEvent.mouseOver(component);
    });
    jest.advanceTimersByTime(1000);
    const contentByName = screen.queryByText(colorNameTooltip);
    const expected = screen.getByText(contentByColorTooltip);

    expect(contentByName).toBeNull();
    expect(expected).toBeTruthy();
  });
});
