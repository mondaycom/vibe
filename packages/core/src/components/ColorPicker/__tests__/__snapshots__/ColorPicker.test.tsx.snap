// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with empty props 1`] = `
<div
  aria-describedby="Pick color"
  aria-labelledby="Color Picker Dialog"
  className="dialogContentContainer colorPicker colorPickerDialogContent typePopover sizeSmall"
  data-testid="dialog-content-container"
  role="dialog"
  style={
    {
      "width": 224,
    }
  }
>
  <div
    style={
      {
        "width": 200,
      }
    }
    tabIndex={-1}
  >
    <ul
      className="colorsGrid"
      data-testid="color-picker"
      role="grid"
      tabIndex={0}
    >
      <li
        className="itemWrapper"
        data-testid="color-picker-item_grass_green"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="grass_green"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-grass_green)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_done-green"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="done-green"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-done-green)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_bright-green"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="bright-green"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-bright-green)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_saladish"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="saladish"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-saladish)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_egg_yolk"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="egg_yolk"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-egg_yolk)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_working_orange"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="working_orange"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-working_orange)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_dark-orange"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="dark-orange"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-dark-orange)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_peach"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="peach"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-peach)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_sunset"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="sunset"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-sunset)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_stuck-red"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="stuck-red"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-stuck-red)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_dark-red"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="dark-red"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-dark-red)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_sofia_pink"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="sofia_pink"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-sofia_pink)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_lipstick"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="lipstick"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-lipstick)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_bubble"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="bubble"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-bubble)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_purple"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="purple"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-purple)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_dark_purple"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="dark_purple"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-dark_purple)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_berry"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="berry"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-berry)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_dark_indigo"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="dark_indigo"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-dark_indigo)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_indigo"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="indigo"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-indigo)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_navy"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="navy"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-navy)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_bright-blue"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="bright-blue"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-bright-blue)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_dark-blue"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="dark-blue"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-dark-blue)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_aquamarine"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="aquamarine"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-aquamarine)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_chili-blue"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="chili-blue"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-chili-blue)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_river"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="river"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-river)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_winter"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="winter"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-winter)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_explosive"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="explosive"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-explosive)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_american_gray"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="american_gray"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-american_gray)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_blackish"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="blackish"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-blackish)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_brown"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="brown"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-brown)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_orchid"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="orchid"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-orchid)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_tan"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="tan"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-tan)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_sky"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="sky"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-sky)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_coffee"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="coffee"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-coffee)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_royal"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="royal"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-royal)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_teal"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="teal"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-teal)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_lavender"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="lavender"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-lavender)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_steel"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="steel"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-steel)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_lilac"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="lilac"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-lilac)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
      <li
        className="itemWrapper"
        data-testid="color-picker-item_pecan"
      >
        <div
          className="feedbackIndicator"
        />
        <div
          aria-label="pecan"
          className="clickable colorItem colorItemSizeMedium disableTextSelection"
          data-testid="clickable"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          style={
            {
              "background": "var(--color-pecan)",
            }
          }
          tabIndex={-1}
        >
          <div
            className="colorIndicatorWrapper"
            style={{}}
          />
        </div>
      </li>
    </ul>
  </div>
</div>
`;
